watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -148,71 +215,216 @@ def eq_flow_mass_comp(blk, j):
+                property_package=m.fs.prop_ro,
+                has_pressure_change=True,
+                pressure_change_type=PressureChangeType.calculated,
+                mass_transfer_coefficient=MassTransferCoefficient.calculated,
+                concentration_polarization_type=ConcentrationPolarizationType.calculated,
+            )
+        else:
+            desal.RO = ReverseOsmosis0D(
+                property_package=m.fs.prop_ro,
+                has_pressure_change=True,
+                pressure_change_type=PressureChangeType.calculated,
+                mass_transfer_coefficient=MassTransferCoefficient.calculated,
+                concentration_polarization_type=ConcentrationPolarizationType.calculated,
+            )
+
+        desal.RO.width.setub(2000)
+        desal.RO.area.setub(20000)
+
+        desal.S1 = Separator(property_package=m.fs.prop_ro, outlet_list=["P2", "PXR"])
+        desal.M1 = Mixer(
+            property_package=m.fs.prop_ro,
+            momentum_mixing_type=MomentumMixingType.equality,
+            inlet_list=["P2", "P3"],
+        )
+        desal.PXR = PressureExchanger(property_package=m.fs.prop_ro)
+        desal.P3 = Pump(property_package=m.fs.prop_ro)
+
+        m.fs.permeate = Product(property_package=m.fs.prop_ro)
+        m.fs.brine = Product(property_package=m.fs.prop_ro)
+
+        # translator blocks
+        m.fs.tb_nf_ro = Translator(
+            inlet_property_package=m.fs.prop_nf, outlet_property_package=m.fs.prop_ro
+        )
+
+        # since the dye << tds: Assume RO_TDS = NF_tds + NF_dye
+        @m.fs.tb_nf_ro.Constraint(["H2O", "dye"])
+        def eq_flow_mass_comp(blk, j):
+            if j == "dye":
+                return (
+                    blk.properties_in[0].flow_mass_comp["dye"]
+                    + blk.properties_in[0].flow_mass_comp["tds"]
+                    == blk.properties_out[0].flow_mass_phase_comp["Liq", "TDS"]
+                )
+            else:
+                return (
+                    blk.properties_in[0].flow_mass_comp["H2O"]
+                    == blk.properties_out[0].flow_mass_phase_comp["Liq", "H2O"]
+                )
+
     else:
-        m.fs.dye_retentate = Product(property_package=m.fs.prop)
+        pass
 
     # connections
-    m.fs.s01 = Arc(source=m.fs.feed.outlet, destination=dye_sep.P1.inlet)
-    m.fs.s02 = Arc(source=dye_sep.P1.outlet, destination=dye_sep.nanofiltration.inlet)
-    m.fs.s03 = Arc(
-        source=dye_sep.nanofiltration.treated, destination=m.fs.permeate.inlet
+    if hasattr(m.fs, "pretreatment"):
+        m.fs.s_feed = Arc(source=m.fs.feed.outlet, destination=prtrt.wwtp.inlet)
+        prtrt.s01 = Arc(source=prtrt.wwtp.treated, destination=dye_sep.P1.inlet)
+        prtrt.s02 = Arc(
+            source=prtrt.wwtp.byproduct, destination=m.fs.wwt_retentate.inlet
+        )
+    else:
+        m.fs.s_feed = Arc(source=m.fs.feed.outlet, destination=dye_sep.P1.inlet)
+
+    dye_sep.s01 = Arc(
```

Comment:
Have you ensured s01is never overwritten? Unclear to me based on the restriction of config combinations.

Reply:
I see some exist on fs and some exist on treatment section blocks though

Reply:
`m.fs.s01` is only created now if there is a GAC or dewatering unit. I believe arcs like `prtrt.s01` are fundamentally different than `m.fs.s01`. I'm not sure if I've addressed your question...